apps/meteor/app/lib/server/functions/saveUserIdentity.ts

```diff
@@ -115,33 +119,36 @@ async function updateUsernameReferences({
 	previousName: string | undefined;
 	rawName?: string;
 	nameChanged: boolean;
+	session?: ClientSession;
 }): Promise<void> {
 	if (usernameChanged && typeof rawUsername !== 'undefined') {
 		const fileStore = FileUpload.getStore('Avatars');
-		const previousFile = await fileStore.model.findOneByName(previousUsername);
-		const file = await fileStore.model.findOneByName(username);
+		const previousFile = await fileStore.model.findOneByName(previousUsername, { session });
+		const file = await fileStore.model.findOneByName(username, { session });
 		if (file) {
-			await fileStore.model.deleteFile(file._id);
+			await fileStore.model.deleteFile(file._id, { session });
 		}
 		if (previousFile) {
-			await fileStore.model.updateFileNameById(previousFile._id, username);
+			await fileStore.model.updateFileNameById(previousFile._id, username, { session });
 		}
 
-		await Messages.updateAllUsernamesByUserId(user._id, username);
-		await Messages.updateUsernameOfEditByUserId(user._id, username);
+		await Messages.updateAllUsernamesByUserId(user._id, username, { session });
+		await Messages.updateUsernameOfEditByUserId(user._id, username, { session });
 
 		const cursor = Messages.findByMention(previousUsername);
 		for await (const msg of cursor) {
 			const updatedMsg = msg.msg.replace(new RegExp(`@${previousUsername}`, 'ig'), `@${username}`);
-			await Messages.updateUsernameAndMessageOfMentionByIdAndOldUsername(msg._id, previousUsername, username, updatedMsg);
+			await Messages.updateUsernameAndMessageOfMentionByIdAndOldUsername(msg._id, previousUsername, username, updatedMsg, { session });
 		}
 
+		// Mongo does not recommend paralelizing session transactions
+		// This is working fine and should be as long as the first transaction is no parallel
 		const responses = await Promise.all([
-			Rooms.replaceUsername(previousUsername, username),
-			Rooms.replaceMutedUsername(previousUsername, username),
-			Rooms.replaceUsernameOfUserByUserId(user._id, username),
-			Subscriptions.setUserUsernameByUserId(user._id, username),
-			LivechatDepartmentAgents.replaceUsernameOfAgentByUserId(user._id, username),
+			Rooms.replaceUsername(previousUsername, username, { session }),
+			Rooms.replaceMutedUsername(previousUsername, username, { session }),
+			Rooms.replaceUsernameOfUserByUserId(user._id, username, { session }),
```

Comment:
is it possible to have these 3 as one single operation? that way the tx reduces load

Reply:
It does make sense, but I think this would be effort for another PR. WDYT?